test/test_case.rb

```diff
@@ -0,0 +1,41 @@
+# typed: true
+# frozen_string_literal: true
+
+# NOTE: This module is intended to be used by addons for writing their own tests, so keep that in mind if changing.
+
+module RubyLsp
+  module TestCase
```

Comment:
I've made this a module rather than a class so that it can used from addons that don't use minitest, such as `ruby-lsp-rspec.`

Reply:
(maybe `RubyLsp::TestHelpers` would be a better name)